hazelcast/src/main/java/com/hazelcast/map/impl/query/AggregationResult.java

```diff
@@ -55,10 +55,14 @@ public AggregationResult(Aggregator aggregator) {
 
     @Override
     public void combine(AggregationResult result) {
+        Collection<Integer> otherPartitionIds = result.getPartitionIds();
```

Comment:
I was thinking about this. but then I saw there were already checks in place and I was worried about side-effects. it's probably something Tom should deal with in 3.9 with this Query engine re-work.